types/query/pagination.go

```diff
@@ -0,0 +1,99 @@
+package query
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/store/types"
+)
+
+// maxLimit is the maximum allowed `limit` for queries
+// if the `limit` is not supplied, paginate will use `maxLimit` as default `limit`
+const maxLimit = 100
+
+// Paginate does pagination of all the results in the PrefixStore based on the
+// provided PageRequest. onResult should be used to do actual unmarshaling.
+//
+// Ex:
+//		prefixStore := prefix.NewStore(store, someRequestParam)
+//		var results []Result
+//		pageRes, err := query.Paginate(accountStore, req.Page, func(key []byte, value []byte) error {
```

Comment:
This is nice, and it would be nicer to have it in the form of a [testable example](https://golang.org/pkg/testing/#hdr-Examples).